### PR TITLE
Add status button tests

### PR DESCRIPTION
**Pull request by vinayak-4**

### CODE REVIEW

File: src/components/StatusButton/__test__/index.test.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+describe("Status button component", () => {
+  it("should render status button correctly", () => {
+    render(<StatusButton />);
+    const statusButton = screen.getByTestId("active-status");
+    expect(statusButton).toBeInTheDocument();
+  });
+
+  it("on hover show status list", () => {
```
Comment (Delwalt) [inline src/components/StatusButton/__test__/index.test.jsx:15]: ~~hover~~ click

File: src/components/StatusButton/__test__/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import StatusButton from "..";
+
+const MockStatusButton = () => {
```
Comment (Delwalt) [inline src/components/StatusButton/__test__/index.test.jsx:4]: ????

File: src/components/StatusButton/__test__/index.test.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    expect(buttonList).toHaveLength(4);
+  });
+
+  it("should not have check icon", () => {
+    render(<StatusButton />);
+    const checkIcon = screen.queryByTestId("check-icon");
+    expect(checkIcon).not.toBeInTheDocument();
+  });
+
+  it("on click should show check icon", () => {
```
Comment (Delwalt) [inline src/components/StatusButton/__test__/index.test.jsx:35]: you can add one more check if value/id is passed as a prop then that value is selected

**Review by Delwalt — CHANGES_REQUESTED** [review]

File: src/components/StatusButton/__test__/index.test.jsx (hunk 1 above)
Comment (vinayak-4) [reply src/components/StatusButton/__test__/index.test.jsx:15]: when a user hovers over the dropdown icon it show the list

File: src/components/StatusButton/__test__/index.test.jsx (hunk 2 above)
Comment (vinayak-4) [reply src/components/StatusButton/__test__/index.test.jsx:4]: I used this function for mocking a render with props when I was testing clicking on a button but it is of no use now. I will remove it.

**Review by Delwalt — APPROVED** [review]

**Pull request merged**